sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/IndexActionConverter.java

```diff
@@ -52,26 +51,42 @@ public static <T> com.azure.search.documents.implementation.models.IndexAction m
         com.azure.search.documents.implementation.models.IndexAction indexAction =
             new com.azure.search.documents.implementation.models.IndexAction().setActionType(obj.getActionType());
 
-        Map<String, Object> mapProperties = IndexActionHelper.getProperties(obj);
-        if (mapProperties == null) {
-            T properties = obj.getDocument();
-            if (serializer == null) {
-                try {
-                    String serializedJson = getDefaultSerializerAdapter().serialize(properties,
-                        SerializerEncoding.JSON);
-                    mapProperties = getDefaultSerializerAdapter().deserialize(serializedJson,
-                        MAP_STRING_OBJECT_TYPE_REFERENCE.getJavaType(), SerializerEncoding.JSON);
-                } catch (IOException ex) {
-                    throw LOGGER.logExceptionAsError(
-                        new RuntimeException("Failed to serialize IndexAction.", ex));
-                }
+        // Attempt to get the document as the Map<String, Object> properties.
+        Object document = IndexActionHelper.getProperties(obj);
+        if (document == null) {
+            // If ths document wasn't a Map type, get the generic document type.
+            document = obj.getDocument();
+        }
+
+        // Convert the document to the JSON string representation.
+        String documentJson;
+        if (serializer == null) {
+            // A custom ObjectSerializer isn't being used, fallback to default JacksonAdapter.
+            try {
+                documentJson = getDefaultSerializerAdapter().serialize(document, SerializerEncoding.JSON);
+            } catch (IOException ex) {
+                throw LOGGER.logExceptionAsError(new UncheckedIOException(ex));
+            }
+        } else {
+            // A custom ObjectSerializer is being used, use it.
+            documentJson = new String(serializer.serializeToBytes(document), StandardCharsets.UTF_8);
+        }
+
+        if (documentJson != null) {
+            boolean startsWithCurlyBrace = documentJson.startsWith("{");
+            boolean endsWithCurlyBrace = documentJson.endsWith("}");
```

Comment:
Is content expected to be trimmed? I.e. is `\t\t{ ...  }\n\n` possible ?

Reply:
It should be trimmed given that it's serialization of the object without passing through other calls.